src/dev/flang/ast/Expr.java

```diff
@@ -531,6 +531,10 @@ boolean needsBoxingForGenericOrThis(AbstractType frmlT)
   private boolean needsBoxing(AbstractType frmlT)
   {
     var t = type();
+    if (frmlT.isThisTypeInTypeFeature())
+      {
+        return false;
```

Comment:
@tokiwa-software/developers I'm not 100% sure this is always the case...

Reply:
Since only values of this type are assignable to this type, that should be ok. What are the actual types when this special case is required?

The whole method, however, is (and was) a fairly ugly collection of single cases.